condensation-forum/application.py

```diff
@@ -301,6 +314,117 @@ def logoutCallback():
+  
+    return redirect(url_for("fileManager"))
+
+@application.route('/fileManager', methods=['GET', 'POST'])
+@authManager.requireAuthentication
+def fileManager():
+    user = authManager.getUserData()
+    if request.method == 'POST':
+        # do stuff when the form is submitted
+        if not user:
+            abort(403)
+        id = user['id']
+
+        if 'file' not in request.files:
+            flash('No file part')
+            return redirect(request.url)
+        file = request.files['file']
+        # if user does not select file, browser also
+        # submit a empty part without filename
+        if file.filename.strip() == '':
+            flash('No selected file')
+            return redirect(request.url)
+        if file:
+            # Determine shortened file name (secure)
+            filename = secure_filename(file.filename.strip())
+            while (len(filename) > 50):
+                cutString = len(filename)%50
+                filename = filename[cutString:len(filename)]
+
+            # Determine the S3 key
+            try:
+                myUuid = uuid.uuid4().hex
+                fn, fileExtension = os.path.splitext(filename)
+                key = id + "/" + myUuid + fileExtension
+                s3client.upload_fileobj(file, bucket_name, key, ExtraArgs={"ACL": "public-read", "ContentType": file.content_type})
+                url = "https://s3-us-west-2.amazonaws.com/condensation-forum/" + key
+                storeToDB(id, url, key, filename)
+                flash('Upload Successfully')
+                return redirect("/fileManager")
+            except Exception as e:
+                print("Something happen: ",e)
+                return e
+
+        else:
+            return redirect("/fileManager")
+        
+        # redirect to end the POST handling
+        # the redirect can be to the same route or somewhere else
+        return redirect("/fileManager")
+    
+    if request.method == 'GET':
+        if not user:
+            abort(403)
+        id = user['id']
+        #Get the user's profile from the DB and zip it first
+        with dataSessionMgr.session_scope() as dbSession:
+            files = query.getFilesByUser(dbSession,id)
+            files = query.extractOutput(files)
+
+        if files != None:
+            fileManagerRendered = fileManagerTemplate.render(files=files)
+            return bodyTemplate.render(
+                title="File Manager",
+                body = fileManagerRendered,
+                user=user,
+                # Todo: Incorporate removeURL here
+                location=request.url)
+        
+    fileManagerRendered = fileManagerTemplate.render()
+    return bodyTemplate.render(
+                title="File Manager",
+                body = fileManagerRendered,
+                user=user,
+                # Todo: Incorporate removeURL here
+                location=request.url)
+
+def storeToDB(userID,url,key,filename):
```

Comment:
This is a 3 line method that is only used once, I think it it makes sense to put this inline

Reply:
I would like to keep this one because it has it's own try catch. I don't want to mess with the posthander try catch